lib/ecto_job/job_queue.ex

```diff
@@ -425,4 +429,30 @@ defmodule EctoJob.JobQueue do
     |> Kernel.+(seconds)
     |> DateTime.from_unix!()
   end
+
+  @spec notify_failed(repo(), job(), Keyword.t()) :: :ok
+  defp notify_failed(_repo, _job = %{notify: nil}, _updates), do: :ok
+
+  defp notify_failed(
+         repo,
+         job = %{notify: _payload},
+         _updates = [state: "RETRY", schedule: _]
+       ) do
+    do_notify_failed(repo, job, "retry")
+  end
+
+  defp notify_failed(
+         repo,
+         job = %{notify: _payload},
+         _updates = [state: "FAILED", expires: _]
+       ) do
+    do_notify_failed(repo, job, "failed")
+  end
+
+  @spec do_notify_failed(repo(), job(), binary()) :: :ok
+  defp do_notify_failed(repo, _job = %queue{notify: payload}, event) do
+    topic = queue.__schema__(:source) <> event
```

Comment:
Should this be `<> “.” <> event` ?

Reply:
Ahh yes, correct it should be.  I have that fixed locally but haven't pushed up to this branch again yet.  Good catch, I'll update this branch now.